src/schema/v2/me/secondFactors/mutations/createSmsSecondFactor.ts

```diff
@@ -0,0 +1,112 @@
+import {
+  GraphQLNonNull,
+  GraphQLString,
+  GraphQLInputObjectType,
+  GraphQLObjectType,
+  GraphQLList,
+  GraphQLUnionType,
+} from "graphql"
+import { mutationWithClientMutationId } from "graphql-relay"
+import { SmsSecondFactor } from "../secondFactors"
+import { ResolverContext } from "types/graphql"
+
+const SmsSecondFactorAttributes = new GraphQLInputObjectType({
+  name: "SmsSecondFactorAttributes",
+  fields: {
+    countryCode: {
+      type: GraphQLString,
+    },
+    phoneNumber: {
+      type: GraphQLString,
+    },
+  },
+})
+
+const ErrorType = new GraphQLObjectType<any, ResolverContext>({
+  name: "Error",
+  fields: () => ({
+    code: {
+      type: new GraphQLNonNull(GraphQLString),
+    },
+    message: {
+      type: new GraphQLNonNull(GraphQLString),
+    },
+  }),
+})
+
+const ErrorsType = new GraphQLObjectType<any, ResolverContext>({
+  name: "Errors",
+  fields: {
+    errors: {
+      type: new GraphQLList(ErrorType),
+    },
+  },
+})
+
+const ResponseOrErrorsType = new GraphQLNonNull(
+  new GraphQLUnionType({
+    name: "SmsSecondFactorOrErrorsUnion",
+    types: [SmsSecondFactor, ErrorsType],
+    resolveType: (data) => {
+      if (data._id) {
+        return SmsSecondFactor
+      }
+      return ErrorsType
+    },
+  })
+)
+
+export const createSmsSecondFactorMutation = mutationWithClientMutationId<
+  any,
+  any,
+  ResolverContext
+>({
+  name: "CreateSmsSecondFactor",
+  inputFields: {
+    password: {
+      type: new GraphQLNonNull(GraphQLString),
+    },
+    attributes: {
+      type: new GraphQLNonNull(SmsSecondFactorAttributes),
+    },
+  },
+  outputFields: {
+    secondFactorOrErrors: {
+      type: ResponseOrErrorsType,
+      resolve: (result) => result,
+    },
+  },
+  mutateAndGetPayload: async (
+    { password, attributes },
+    { createSecondFactorLoader }
+  ) => {
+    if (!createSecondFactorLoader) {
+      throw new Error("You need to be signed in to perform this action")
+    }
+
+    try {
+      const snakeCaseAttributes = {
+        country_code: attributes.countryCode,
+        phone_number: attributes.phoneNumber,
+      }
+
+      const data = await createSecondFactorLoader({
+        kind: "sms",
+        password,
+        attributes: snakeCaseAttributes,
+      })
+
+      return data[0]
+    } catch (error) {
+      const { body } = error
```

Comment:
To abide by the format of these mutation errors (slightly different than our preferred pattern), I'll just drop in a simple helper to format as needed.